src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,192 @@
+    {
+        private const string ResultsFileArgumentName = "results-file-name";
+        private const string ResultsPathArgumentName = "test-results-path";
+        private const string SummaryMessageArgumentName = "test-execution-summary";
+        private const string ExitCodeArgumentName = "return-code";
+        private const string ExcludeCategoriesDirArgumentName = "exclude-categories-dir";
+        private const string ExcludeCategoriesFileArgumentName = "exclude-categories-file";
+        private const string ExcludeMethodArgumentName = "exclude-method";
+        private const string IncludeMethodArgumentName = "include-method";
+        private const string ExcludeClassArgumentName = "exclude-class";
+        private const string IncludeClassArgumentName = "include-class";
+
+        private Bundle? _bundle;
+
+        protected Dictionary<string, string> bundleArguments = new Dictionary<string, string>();
+
+        protected AndroidInstrumentationBase(IntPtr handle, JniHandleOwnership transfer)
+            : base(handle, transfer)
+        {
+        }
+
+        protected virtual IEnumerable<Assembly> Tests
+        {
+            get
+            {
+                yield return Assembly.GetExecutingAssembly();
+            }
+        }
+
+        public override void OnCreate(Bundle? arguments)
+        {
+            base.OnCreate(arguments);
+
+            _bundle = arguments;
+
+            Start();
+        }
+
+        public override async void OnStart()
+        {
+            base.OnStart();
+
+            var bundle = new Bundle();
+
+            var entryPoint = new TestsEntryPoint(_bundle);
+            entryPoint.TestsCompleted += (sender, results) =>
+            {
+                var message = String.Format("Tests run: {0} Passed: {1} Inconclusive: {2} Failed: {3} Ignored: {4}",
+                results.SkippedTests, results.ExecutedTests, results.PassedTests, results.InconclusiveTests, results.FailedTests);
+
+                bundle.PutString(SummaryMessageArgumentName, message);
+                
+                bundle.PutLong(ExitCodeArgumentName, (long)(results.FailedTests == 0 ? ExitCode.SUCCESS : ExitCode.TESTS_FAILED));
+            };
+
+            entryPoint.Tests = Tests;
+
+            await entryPoint.RunAsync();
+
+            if (File.Exists(entryPoint.TestsResultsFinalPath))
+            {
+                bundle.PutString(ResultsPathArgumentName, entryPoint.TestsResultsFinalPath);
+            }
+            
+            if (bundle.GetLong(ExitCodeArgumentName, 82) == (long)ExitCode.RETURN_CODE_NOT_SET)
+            {
+                bundle.PutLong(ExitCodeArgumentName, (long)ExitCode.RETURN_CODE_NOT_SET);
+            }
+
+            Finish(Result.Ok, bundle);
+        }
+
+        class TestsEntryPoint : AndroidApplicationEntryPoint
+        {
+            readonly string _resultsPath;
+            readonly string _excludeCategoriesDir;
+            readonly string _excludeCategoriesFile;
```

Comment:
I think @premun  whats a 'private' in those:
```suggestion
            private readonly string _resultsPath;
            private readonly string _excludeCategoriesDir;
            private readonly string _excludeCategoriesFile;
```